### ISSUE TITLE
Can't send MMS with Bouygues Telecom

### ISSUE THREAD

**Issue by gbdomubpkm** [HelloVolla/ubuntu-touch-beta-tests#94]
RC 8
Bouygues SIM (for me)

Not possible to send MMS (spinning wheel) and to receive MMS using messaging app.

The command line
```/usr/share/ofono/scripts/set-context-property 0 Protocol IP```
fixed internet (cellular data ok) but not MMS.
APN MMS is ok.
But since I hadn't tested the MMS function before the fix, I can't say more.
Sms, calls are ok.

**Comment by JamiKettunen** [comment]
I can confirm this issue on Development channel build 2021-02-22 as well, MMS receiving works fine, but trying to send MMS just shows the following log in `~/.cache/upstart/nuntium.log`:
```
2021/02/27 14:15:22 cannot create new ContentType for image20210225_031829661.jpg of content type image/jpeg on : open : no such file or directory
```
Looking at the [related code of nuntium](https://github.com/ubports/nuntium/blob/54bab54/mms/attachments.go#L61) it appears `filePath` for the attachment is empty for some reason; I've brought this up in the UBports development group.

Also this might be new generic issue tracked on upstream @ https://github.com/ubports/ubuntu-touch/issues/1670.

**Comment by wurzer** [comment]
Did you check the APN settings for MMS? This works fine for me with OTA 15 Stable

**Comment by gbdomubpkm** [comment]
@wurser You have probably right (but i didn't test) for OTA-15 as it's about RC and DEV currently. But if the problem isn't fixed, you could reproduce for OTA-16, probably.. 😉

**Comment by gbdomubpkm** [comment]
@UniversalSupetBox Thank you so much for the fix. I could test it in RC 10.

**Comment by gbdomubpkm** [comment]
@UniversalSuperBox
Sorry to say that but it doesn't work for me in RC 10 yet. But there is a progress : it says failure after the  spinning wheel ! Same thing for little audio or picture file.
![screenshot20210303_203121379](https://user-images.githubusercontent.com/50362604/109862378-a876e180-7c60-11eb-8333-5ce92f27c745.png)
I tried to send  to myself or somebody else.

**Comment by JamiKettunen** [comment]
I was able to send an MMS message succesfully under Development channel build 2021-03-04, so I'll go ahead and close this now :) @gbdomubpkm's issue appears to be something else as it also happens on OTA-15 and I'm working with them directly to attempt resolving this.

**Comment by gbdomubpkm** [comment]
@UniversalSuperBox

For info as it could be useful for you : RC 10 : my operator (and why not others!): the MMS context cannot be activated (shown in nuntium log)

![cannot set context to true](https://user-images.githubusercontent.com/50362604/110197947-18e35580-7e4f-11eb-8b8f-0f8a33f33884.png)

And thanks again to @JamiKettunen for all the answers he gives me to THE problems I bring to him !!

**Comment by gbdomubpkm** [comment]
![screenshot20210311_182927874](https://user-images.githubusercontent.com/50362604/110829510-7edd3c00-8298-11eb-9ed1-5059cab8421f.png)

**Comment by gbdomubpkm** [comment]
OTA-16: no MMS confirmed alas. This is the black spot as MMS is a fundamental (for me).

**Comment by wurzer** [comment]
I use MMS successfully on my Ubuntu Touch device with OTA 15 of the stable channel. Please check your APN settings. You can type "MMS" on our help page at https://volla.online/en/faqs/
In some cases a reboot is necessary to get the MMS feature working

**Comment by gbdomubpkm** [comment]
The problem is real, and was identified by @NotKit on March 25 using the logcat_radio.txt that I provided him and captured during the error sending an MMS.
No info since.

**Comment by wurzer** [comment]
We need to be very precise to find a solution.

I can confirm, That cellular data connection is working as soon your SIM is activated in a Volla Phone with Ubuntu Touch OTA-16 (stable channel). I some cases you need to enter the APN setting for internet and MMS yourself. 

If the APN settings are not working instantly, you may need to reboot. Then you can send and receive MMS. 

We need to find our, what's different on your device. 

- A known issue is, that you can use the 2nd SIM Card for data connection. We are working on a fix, that probably come with OTA-17. 
- Take care, that you set the checkmarks for activating the APN settings. 

We need to find our, what's different to your and my conditions on the Volla Phone.

![screenshot20210424_113235726](https://user-images.githubusercontent.com/7789612/115954386-08676700-a4f1-11eb-8690-ad11955dd4cd.png)
![screenshot20210424_113258090](https://user-images.githubusercontent.com/7789612/115954388-09989400-a4f1-11eb-9244-37cdc91dc888.png)
![screenshot20210424_113305249](https://user-images.githubusercontent.com/7789612/115954389-09989400-a4f1-11eb-9c27-9a48935890f5.png)
![screenshot20210424_113312390](https://user-images.githubusercontent.com/7789612/115954390-0a312a80-a4f1-11eb-8c66-7756b3d7c63f.png)

**Comment by gbdomubpkm** [comment]
Hello @wurzer 
At my level all the obvious settings and what is possible has already been tested for a long time.
As already shown above, the MMS context cannot be activated (even if it seems to be set up in appearance).


So much the better if the MMS works for a majority of SIM categories and if the MMS works with your SIM.

As far as the SIM I own is concerned, I can confirm that MMS does not work on the STABLE channel, nor on the RC 13 channel tested today.

This is not a problem of my own SIM, but probably related to a category of SIM for which MMS does not currently work on the Volla (or on some Halium 9 devices ? I don't know).

NotKit will be able to confirm and enlighten you better than me on the nature of the malfunction(s) that I do not know exactly even if i gave him the logcat_radio.txt file referencing the MMS Error.

**Comment by wurzer** [comment]
> Hello @wurzer
> At my level all the obvious settings and what is possible has already been tested for a long time.
> As already shown above, the MMS context cannot be activated (even if it seems to be set up in appearance).

In which country are you located and which carrier do you use?

**Comment by gbdomubpkm** [comment]
It is written at the beginning.... Bouygues SIM
France

**Comment by gbdomubpkm** [comment]
This annoying at this day  unresolved issue  is no more personal than other currently open bug reports do. This issue should be reopened (i can't do myself) as long as the situation remains as it is.

**Comment by wurzer** [comment]
I can understand, that missing MMS feature is a critical issue. The challenge for my colleagues and me is, that we can't reproduce this issue. I can send and receive MMS with the correct APN settings. I assume, that we need a sIM of your carrier in this specific french network to reproduce the problem. We might need your ofono log or a community developer, who is able to reproduce the problem. Are you familiar with Ubuntu Touch logging?

**Comment by gbdomubpkm** [comment]
Already mentioned: different logs have already been communicated. I couldn't do more. Only test a solution by being guided step by step if necessary. I am not a developer.

**Comment by NotKit** [comment]
My current understanding of the problem is that particular carrier uses a single APN for both internet context and MMS (mmsbouygtel.com). Older ofono version used on Nexus 5 and other pre-Android 9 devices allowed to configure Internet + MMS type APN, but SailfishOS fork of ofono we use instead has hardcoded assumptions about there being a single APN context for internet and a separate one for MMS.

@gbdomubpkm also tried to configure separate different APNs for internet and MMS, but second context activation still fails on Volla Phone in this case. I suspect carrier may allow only one active context in this case, and older devices may workaround it by shortly disconnecting from Internet context.

I need to look more into ofono code we use to see if the assumption about contexts can be removed. If yes, this should also fix the problem we have with adding/removing contexts from Ubuntu Touch Settings UI.

**Comment by gbdomubpkm** [comment]
Finally and after many tests, with a different APN for INTERNET (ebougtel.com) and for MMS (mmsbouygtel.com) and by activating (by linux command line) after each reboot of the Volla the context 2 MMS by being at the beginning on the type of connection 2g/3g with a 3g connection, then by going on the type of connection 2g/3g/4g with the 4g connection and by reactivating again the context 2 MMS, the MMS function work in 4g in the limit of this procedure and other limits.. It's a far from a perfect solution or a normal fix  but better than nothing currently.

**Comment by wurzer** [comment]
I can confirm, that I also can use two SIM cards:

- I can select both SIM cards for mobile data access, but as expected, only one can be activated at a time. 
- I can switch between the two SIM cards and the device will switch data access after some seconds.
- I can send and receive MMS with both SIM cards, **but only with the activated one.** 
- I can send and receive SMS with bosh SIM cards independent from the selected one for data access
- Receiving and making calls is no problem at all with two SIM cards.

It took indeed at least one reboot to have the setting properly working.

**Comment by gbdomubpkm** [comment]
Hello.
Any progress in solving the problem?

**Comment by wurzer** [comment]
MMS works fine for most carrier and use cases. 

Only two issues are known:

1. If you have two SIM cards, Receiving and sending MMS is only possible for SIM card that is activated for data access.
2. If the carrier has the same APN settings for data and MMS, Ubuntu Touch can't handle it right now.

We are working on both issues. 

In any case I strongly recommend, not to use command line tools to configure network access but only the system setting sof Ubuntu Touch.

**Comment by Pekkari** [comment]
Hi, would it be possible for you also to test the following steps I sent
on [this comment](https://github.com/HelloVolla/ubuntu-touch-beta-tests/issues/105#issuecomment-931539017)?

Thanks!

**Comment by gbdomubpkm** [comment]
Hello @Pekkari
I took some times because i did some tests.
I installed PR-28 directly with the phone on OTA-19.

BUT i have good news 🙂. It works !

I managed to activate a single APN (internet + mms) by modifying a proposed APN after reboot, but not automatically as you thought.

The single APN is retained after reboot, which is also good news.

I don't have any more MMS context disabling if I activate flight mode and then deactivate it, or after a reboot.

I had trouble activating cellular data (internet) 'connection types' 2G and 2G/3G but in fact it seems to work and so does MMS.

It now seems possible to delete unnecessary APNs which is good news even if they tend to reappear (but not activate) after a reboot.

@NotKit and you did good work 👍👋

If i see something new, i will tell you.

**Comment by Pekkari** [comment]
Very good, thanks for the testing, and the patience. I'm certain we'll work in improving the apn area of
system settings, and we'll certainly work in upstreaming those changes to ubports to let you have them
later on through OTA instead of ubports-qa.

**Comment by gbdomubpkm** [comment]
Yes. Ideally an update for OTA-20 would be nice.
What would happen if I switched to RC 34 ?. Would I lose the PR? But i wouldn't do it because RC-34 was 'morph' buggy for me on the volla 😉

**Comment by wurzer** [comment]
I highly recommend to keep the stable channel. other channels are automatically triggered build, which could contain errors. These builds are for developers and beta testers. 

We expect hte OTA-20 update next month at least this year.

**Comment by gbdomubpkm** [comment]
@Pekkari Hello.
I'm not making currently a cause and effect relationship, as I don't know when it started exactly, but I don't have any SOUND notifications under dekko mail with OTA-19 and Volla smartphone, only the ENVELOPE notifications.
Can you check with your Volla?
Thank you.

**Comment by Pekkari** [comment]
I'm afraid no, I'm on a sick leave, I can't work for now on this.

**Comment by gbdomubpkm** [comment]
Okay. I was right to ask. First, I wish you a very good recovery. Will your actual MMS command line procedure work well on OTA-20 ? As it is, I see no point in switching to OTA-20. OTA-19 works very well as it is and is in balance. Webgl also works very well. I hope that this mechanism will be present on OTA-21.

**Comment by wurzer** [comment]
I think your problem is related to the case, that the APN settings of network and MMS are the same. We are working on that issue, but it's not ready for a pull request, yet. I'm confident to have solution for OTA-21 this year. We already have a solution, but it's not stable yet.

**Comment by gbdomubpkm** [comment]
Hello @wurzer. As i said before in my comments, https://github.com/ubports/ofono/pull/28 works well with OTA-19, that's why i prefer to stay on OTA-19 currently in case the procedure doesn't work so well on OTA-20.

**Comment by gbdomubpkm** [comment]
Hello again @Pekkari 
i forgot to ask you this thing : In your advisory opinion, can I do (not yet done.. 😉 ) the procedure  [here](https://forums.ubports.com/topic/6510/mms-users-call-for-testing/40) over PR-28 without risk of compromising my balance??

**Comment by Pekkari** [comment]
Not right now, it will break your ofono, as commented right before.

**Comment by gbdomubpkm** [comment]
Okay, thank you for confirming again to stay put!

**Comment by Pekkari** [comment]
hey @gbdomubpkm , check here [this comment](https://github.com/ubports/ofono/pull/28#issuecomment-968288158), you may be able to restore your current mms context if you do some further testing, I pushed the buggy code to another branch to let community continue testing.

**Comment by gbdomubpkm** [comment]
ok thanks @Pekkari  : you restored the PR-28 that was working if i understood well. But what does it mean exactly ? this PR-28 :  should work with OTA-20 only ? Should work with OTA-20  AND https://forums.ubports.com/topic/6510/mms-users-call-for-testing/40 too ?

**Comment by Pekkari** [comment]
As long as the CI system doesn't say otherwise, it should work with any becoming OTA, so feel free to upgrade, and retest.

**Comment by gbdomubpkm** [comment]
Hello @Pekkari 
I'm on OTA-20... and...
![screenshot20211121_082229647](https://user-images.githubusercontent.com/50362604/142753566-c8af94fb-37f4-4bc8-b5cc-9d9cbc1910fd.png)
It seems not possible currently to restore mms context, ofono 28 failed to built ! 🙂

**Comment by gbdomubpkm** [comment]
Going back to the alternative solution with two different apn, one for internet and the other for mms, which allowed me to send mms in 4g :  I can now with OTA-20 only send mms in 2g, 3g, but the solution does not work anymore in 4g !

**Comment by Pekkari** [comment]
Yep, there was conflicts because we landed another fix, but the build succeded and it was tested against the ci.
Weird that you couldn't at first. We didn't know 4G mms didn't work. I'll transmit the new internally to see what we
can help. Please be patient with that tough.

**Comment by gbdomubpkm** [comment]
> We didn't know 4G mms didn't work

Again, mms doesn't work NOW with my alternative solution (2 different apn) and OTA-20. MMS worked well in 4g with your PR-28 and OTA-19, or could work with alternative solution (2 different apn) and OTA-19 or OTA-18..

**Comment by gbdomubpkm** [comment]
I have good news 🙂 the PR-28 has been installed !! 

But i had an idea : i tried to install the PR under 3g network, NOT 4g and it worked !! Strange thing ! 

But now, if i am under 4g, i can send again mms !! Strange again 🙂

**Comment by gbdomubpkm** [comment]
I haven't tested it yet. If everything works the way you want, should I get this that is to say only one APN ?
![screenshot20211206_153349240](https://user-images.githubusercontent.com/50362604/144866277-afa07cc8-c5fa-41ed-8afb-dea6bb8136ab.png)

**Comment by Pekkari** [comment]
Yes, the expected behaviour is that you'll be able to reboot the device or restart ofono, without
having an extra apn only for mms, it should stay as you show in the pic.

**Comment by gbdomubpkm** [comment]
I'll try it in the evening after work. See you soon.

**Comment by gbdomubpkm** [comment]
Hello.

This version ofono 28 does not work for me 😞😉

- After the 1st reboot, I still got 2 APNs, the good one (INTERNET + MMS) activated, the second one (MMS) not activated.  But the mms didn't work anymore, and I didn't have access to the gallery which didn't open anymore (stuck on blue screen with montains, very nice !). I could attach a picture using file manager but the mms din't work anymore.
- The second time I rebooted, it was even worse: a bad APN was displayed (Internet contract pro data) as at the begining of the mms issue, but the internet still worked.

I tried several times to change, delete the APNs and reboot but nothing worked properly: at best, I had internet but not always.

I've seen that ofono 28 has installed some new things: a new presentation of the unlocking and a new presentation of the storage, for what I've seen.

What do I do now ??? I reflashed the smartphone with the good ofono 28 (you have always it 😲 ?) that worked ??

**Comment by Pekkari** [comment]
something may have gone wrong here, this modifies ofono, it shouldn't modify the UI,
unless it is downgrading it for using old patches. The new patch shouldn't affect the
former behaviour, only improve it, in principle using ofono 28 should be safe since
the behaviour does the default if the settings are not in place. I should suggest to try
to remove and reinstall ofono 28, grant to have only the context you want(plus ims
that doesn't show in the UI), and then test reboot/restart of ofono.

**Comment by gbdomubpkm** [comment]
Last night: I removed the new ofono-28 and it was still a mess. Nothing worked: gallery blocked, new storage, apn problem.
Then I reflashed OTA-20  without wipe: same problems with the new storage and blocked gallery.
So after, I reflashed with wipe option: back to a clean OTA-20 conf and the Volla works normally (with the initial MMS problem).
I went to bed at 2 am.. And i must reinstall other things yet.
Can you give me back the ofono 28 conf that works? I would like to reinstall it.. Thanks

**Comment by gbdomubpkm** [comment]
@stekalb and @spinnster, maybe it will work for you.....

**Comment by Pekkari** [comment]
Unfortunate to hear that, though I managed to restart ofono without having the extra mms
context, so I should hear more from the testing community before ditching this. Thing is that
with all it costed to reach this, I feel this is a last resort to try to solve the problem, so it works,
or we'll need to stay with an incomplete solution by reverting my last patch.

**Comment by gbdomubpkm** [comment]
Hello @Pekkari . Thank you for all your work.
I did my part of tests very late... I can only wait for confirmation or not of the problem I encountered, from other testers for the moment.

**Comment by Pekkari** [comment]
it can help, since it won't be pristine installation per se, but closer to it. Pristine installation
with the patchset would be perfect test, but since you got a bad experience I don't want to
ask you do so.

**Comment by gbdomubpkm** [comment]
Ok, I decided to make a final attempt as clean as possible from my clean OTA-20 too.

- I removed the SIM card, booted the smartphone and installed PR-28 on wifi.
- I turned off the smartphone, installed the SIM card and restarted the smartphone.

Alas, I confirm my previous attempt, that is, this ofono 28 does not work or very poorly at best for me.

The inconveniences and annoyances (new unlocking screen, new storage presentation) are the same with a few nuances: the gallery did not block and I managed to install once the right configuration (internet + mms) and send mms. But after reboot, the configuration was lost. At some point, the apn can no longer be deleted by the trash.

Uninstalling ofono has maintained the new lock screen and the new storage layout.

I will flash UT again with wipe and spend a few hours reinstalling everything.

**Comment by Pekkari** [comment]
I was in ota-19 and upgraded to ota-20, and tried the ubports-qa, it seems to be doing some sort of
brutal downgrade that literally borked the full thing. I have to look for some other way to distribute the fix.
I'll be back.

**Comment by gbdomubpkm** [comment]
A little digression : please, as you just flashed OTA-20, may you do a little test and try to install notch-hax ? It doesn't want to work for me, patch not found. Thanks.

**Comment by Pekkari** [comment]
@gbdomubpkm mind to test to modify  the service provider database under:
`/usr/share/mobile-broadband-provider-info/service-provider.xml`
And make your provider look like:
```
        <provider>
                <name>Bouygues Telecom</name>
                <gsm>
                        <network-id mcc="208" mnc="20"/>
                        <network-id mcc="208" mnc="21"/>
                        <network-id mcc="208" mnc="88"/>

                        <voicemail>660</voicemail>
                        <balance-check>
                                <dtmf>630</dtmf>
                        </balance-check>

                        <apn value="a2bouygtel.com">
                                <plan type="postpaid"/>
                                <usage type="internet"/>
                                <name>Contrat Pro Data Illimité</name>
                        </apn>
                        <apn value="b2bouygtel.com">
                                <plan type="postpaid"/>
                                <usage type="internet"/>
                                <name>B2Bouygtel</name>
                        </apn>
                        <apn value="ebouygtel.com">
                                <plan type="postpaid"/>
                                <usage type="internet"/>
                                <name>Forfait Data</name>
                        </apn>
                        <apn value="mmsbouygtel.com">
                                <plan type="postpaid"/>
                                <usage type="internet"/>
                                <name>Forfait Data</name>
                                <mmsc>http://mms.bouyguestelecom.fr/mms/wapenc</mmsc>
                                <mmsproxy>62.201.129.226:8080</mmsproxy>
                        </apn>
                        <apn value="pcebouygtel.com">
                                <plan type="postpaid"/>
                                <usage type="internet"/>
                                <name>Contrat Pro Data</name>
                        </apn>
                </gsm>
        </provider>
```
You may need to have the so called pr-28 for ofono, and then reset your apns in the celular
settings -> carrier & apn -> apn. After that ensure you have internet active, and try to send
some mms. If it works, I'll modify the database upstream.

Thanks!

**Comment by gbdomubpkm** [comment]
hello.
Some questions before doing anything again.

Service-provider.xml is this currently on my Volla:

	<provider>
		<name>Bouygues Telecom</name>
		<gsm>
			<network-id mcc="208" mnc="20"/>
			<network-id mcc="208" mnc="21"/>
			<network-id mcc="208" mnc="88"/>

			<voicemail>660</voicemail>
			<balance-check>
				<dtmf>630</dtmf>
			</balance-check>

			<apn value="a2bouygtel.com">
				<plan type="postpaid"/>
				<usage type="internet"/>
				<name>Contrat Pro Data Illimité</name>
			</apn>
			<apn value="b2bouygtel.com">
				<plan type="postpaid"/>
				<usage type="internet"/>
				<name>B2Bouygtel</name>
			</apn>
			<apn value="ebouygtel.com">
				<plan type="postpaid"/>
				<usage type="internet"/>
				<name>Forfait Data</name>
			</apn>
			<apn value="mmsbouygtel.com">
				<plan type="postpaid"/>
				<usage type="internet"/>
				<name>Forfait Data</name>
			</apn>
			<apn value="pcebouygtel.com">
				<plan type="postpaid"/>
				<usage type="internet"/>
				<name>Contrat Pro Data</name>
			</apn>
			<apn value="mmsbouygtel.com">
				<usage type="mms"/>
				<name>Bouygues MMS</name>
				<mmsc>http://mms.bouyguestelecom.fr/mms/wapenc</mmsc>
				<mmsproxy>62.201.129.226:8080</mmsproxy>
			</apn>
		</gsm>

Would you like me to change (what in) the bouygues settings in the service-provider file on my phone ? What would be the complete procedure to do this and install this new file IF I DECIDED to do it ? And after I reinstall the latest version of the PR-28 which is not currently working? Anyway, I'm not keen on doing this again, I'm tired of reflashing (with wipe) my smartphone (4 flashes in 15 days) partly also because of the conflicts with the user interface as the gallery might still not work again and that is a problem. Also, I don't see how this would solve the issues of other users who have problems with their APN and with this new PR-28.

The previous PR-28 worked well for me (and other users i think) with few side effects.

**Comment by Pekkari** [comment]
the process is backup the file, and replace the provider entry with the one I provided you. Now
that without the pr 28 is going to help, but not solve the problem, I'm afraid. I'm giving you  the
pill to cure yourself, you decide to take the treatment, or not. Former pr 28 is not complete, so
it is not an option to revert it, and there is users who make it with the new one.

**Comment by gbdomubpkm** [comment]
> replace the provider entry with the one I provided you.

 Linux command lines procedure please to do this and procedure to go back if things went wrong..
I need this to assess the situation.
Thanks.

**Comment by gbdomubpkm** [comment]
I limited myself to make an attempt in RC 44 by installing the new serviceproviders.xml file, but without the current version of the PR-28. I can indeed confirm that this did not solve the problem and even temporarily worsened it: Impossible to register APNs or these appearing totally empty after reboot. I reinstalled the original serviceproviders.xml file.

**Comment by GreUms78** [comment]
Hey,

Its working for me with this settings
Change APN NAME : b2bougtem.com
My update channel is RC
I don't know if it works with OTA 20
Test with OnePlus 6


(Sorry for my english)



![1](https://user-images.githubusercontent.com/97088020/148212114-910842b2-b039-4325-b1c8-843860385aaf.png)
![2](https://user-images.githubusercontent.com/97088020/148212120-b48c140f-9d9c-4b3d-a757-74fe21564654.png)
![3](https://user-images.githubusercontent.com/97088020/148212121-2a22b3af-c4f5-4a88-af9c-fc9fd6b09cf5.png)

**Comment by GreUms78** [comment]
I have completely reinstalled the phone (it's the ota 21)
I changed the APN for MMS and everything works

**Comment by gbdomubpkm** [comment]
Hello @GreUms78
Thank you for your feedback even if we speak of Volla here as it is a Volla issue.

I didn't test on Volla your APNs  but i see something strange for MMS Apn : access point name b2bouygtel.com and MMSC http://mms.bouyguestelecom.fr/... I didn't think it could be different in the same MMS apn @Pekkari.

What do you mean by i changed the APN for MMS ?

**Comment by GreUms78** [comment]
Hey @gbdomubpkm 
I change "Nom du point d'acces" 
"mmsbouygtel.com" by "b2bouygtel.com"

**Comment by Pekkari** [comment]
> Hello @GreUms78 Thank you for your feedback even if we speak of Volla here as it is a Volla issue.
> 
> I didn't test on Volla your APNs but i see something strange for MMS Apn : access point name b2bouygtel.com and MMSC http://mms.bouyguestelecom.fr/... I didn't think it could be different in the same MMS apn @Pekkari.

I don't see why you bind the apn name with the http address, it can
be that changing the name, the rest of settings are populated from a
different entry in the service provider xml, but I haven't tested this
behaviour myself, I can't tell.

**Comment by GreUms78** [comment]
My Settings
![screenshot20220106_130611251](https://user-images.githubusercontent.com/97088020/148385979-206b0922-e68c-44da-a168-b9f4076e0980.png)
![screenshot20220106_130619300](https://user-images.githubusercontent.com/97088020/148385992-9a852937-dad4-4287-8652-5a7ac2b5f082.png)
![screenshot20220106_130628517](https://user-images.githubusercontent.com/97088020/148386018-79aad10c-c266-497f-879c-084b841ea8d3.png)

**Comment by gbdomubpkm** [comment]
Hello @Pekkari 
Hello.
So I finally tested your new serviceproviders.xml file with the PR-28 on OTA-21: it doesn't work, because in fact no APN configuration for MMS wants to register (the MMS APN is presented totally empty in its fields): if I try to create a unique APN INTERNET + MMS complete, the APN registers only for the internet: only mmsbouygtel.com remains; an APN only for MMS remains empty after validation. In fact, this seems logical to me because I don't see anything specified as MMS configuration in your new serviceproviders.xml file.
So the MMS APN is always created empty after reboot even if it is deleted before reboot.
Finally, once again, your first version of the PR-28 was working fine in a stable way with the original serviceproviders.xml file even if there was an additional APN created but not activated (this time it is created but empty!).

I didn't have to reflash my Volla this time: I uninstalled the PR-28 and reinstalled the original serviceproviders file: everything seems to be fine.

**Comment by Pekkari** [comment]
this is copied from the entry I posted you:
```
                        <apn value="mmsbouygtel.com">
                                <plan type="postpaid"/>
                                <usage type="internet"/>
                                <name>Forfait Data</name>
                                <mmsc>http://mms.bouyguestelecom.fr/mms/wapenc</mmsc>
                                <mmsproxy>62.201.129.226:8080</mmsproxy>
                        </apn>
```
This entry indeed, is an internet+mms entry, you have the mms center and proxy defined.
Unfortunate that it didn't work though. I did my best guessing it.

**Comment by gbdomubpkm** [comment]
> <usage type="internet"/

I don't know if it is the reason, but it is probably understood for internet only.

Just an idea and what i write is probably wrong but is it possible to write something like
`usage type=''internet"/"mms"` on the same line ?

**Comment by Pekkari** [comment]
No, the code only understand internet or mms, internet+mms or derivative is not an ofono
definition, but an UI definition, the dferentiator is that an internet apn is not meant to have
mmsc, or mmsproxy, which this have, and the UI *should* interpret it as internet + mms.

**Comment by gbdomubpkm** [comment]
Couldn't you make a fork of your PR-28 corresponding to the one of the end of september 2021, titling it PR-XX and still remain as is and manually installable at leisure? This would also allow to check that it still works as well and maybe better with the other ofono improvements. This would not prevent you from continuing your work if needed on the current PR-28 which would still be testable.

**Comment by Pekkari** [comment]
That I did once as a favor, and now that I have more code, I need more testing and more comments, the only
one who have problems come to ask for doing it again. Please mind that I can't do it forever, I need the new
work to be tested, and confirmed also, because the former solution was *incompleted*.

**Comment by gbdomubpkm** [comment]
I retested PR 28... 😰
Actually, I have some news that is now quite good 🙂: the unique INTERNET + MMS APN with mmsbouygtel.com and other settings can be installed and works fine with the new PR-28 and the original serviceproviders.xml file under OTA-21 !

This configuration is installed from the Bouygues MMS APN that I changed to INTERNET + MMS and that appears not activated (unchecked) at the beginning, and by deleting the Internet APN CONTRAT PRO DATA UNLIMITED that IS activated at the beginning.

BUT, this unique INTERNET + MMS configuration remains after 1 reboot only without creating other APNs : that is good.

At the second reboot, I fall back on the activated INTERNET APN CONTRAT PRO DATA UNLIMITED that i delete again and on the deactivated (unchecked) MMS APN that I can activate and modify again in INTERNET+ MMS APN and it works again !

**Comment by Pekkari** [comment]
Humm... exactly as I was finding it, the new code works once and then it resets the apns. I tested also yesterday
with same result using ubports-qa version instead of my own ofono packages. It's going to be tricky to get that
mms apn out of the boot of ofono, but I have to give it a try. Thanks for testing, and the patience along!

**Comment by gbdomubpkm** [comment]
FYI: although the previous version of the PR-28 was imperfect, as far as the unique INTERNET + MMS APN was concerned, which also worked, it remained whatever the number of reboots.

**Comment by gbdomubpkm** [comment]
About the currenr PR-28, It's not even that simple. After each first reboot, even if the unique APN INTERNET + MMS is still apparent, the sending of MMS does not always work (well); After sending MMS to myself, I noticed that it can also happen that there is a significant time lag between the transmission and the reception of the MMS, as if the MMS remained stuck somewhere, or it does not arrive before sending a second MMS, or not possible to send a MMS at all.
I also noticed that if I rebooted having previously put the flight mode, the cellular data could not be restored, I do not know if it is the same thing by cutting only the cellular data before rebooting...
In short, it may require several reboots to have a really stable internet and MMS working currently.

**Comment by gbdomubpkm** [comment]
Hello @Pekkari 
The last  PR-28 i tested 27 days ago on OTA-21 works 'relatively' well.. Did you update it recently **in a better way**  or it is always the same ?

**Comment by Pekkari** [comment]
> Hello @Pekkari The last PR-28 i tested 27 days ago on OTA-21 works 'relatively' well.. Did you update it recently **in a better way** or it is always the same ?

No, no change from my end, that is visible in the PR.

**Comment by gbdomubpkm** [comment]
Ok.  Thank you. To clarify, I wasn't suggesting that it was working less well now: it was just to see if you've been able to (or are about to) fix what's wrong or make some further improvements.

**Comment by embeddedmz** [comment]
@GreUms78  Thank you very much for the solution ! Now I can receive MMSes on my Xiaomi Mi A2.

**Comment by gbdomubpkm** [comment]
@Pekkari Hello. I haven't yet flashed the latest RC which is the test version of OTA-22, but do you know in principle whether PR-28 should be adversely affected by changes since OTA-21?

**Comment by Pekkari** [comment]
Only in the case new changes for ofono come in the OTA, and I'm not aware of such a thing.

**Comment by gbdomubpkm** [comment]
So I am on RC test OTA-22. The PR-28 seems to work like on OTA-21: the freshly installed single MMS + INTERNET configuration works naturally once, it reappears after reboot but has to be revalidated to work; on the second reboot, the single configuration has to be completely reconfigured (two APNs visible again).

**Comment by GreUms78** [comment]
> @GreUms78 Thank you very much for the solution ! Now I can receive MMSes on my Xiaomi Mi A2.

;) you're welcome

**Comment by gbdomubpkm** [comment]
FYI: in RC 54 (OTA-23 test version) the PR-28 is functional as usual (I'm just saying it works).

**Comment by gbdomubpkm** [comment]
Hello.
I don't use it yet but maybe later I don't know when and I have a question :  although not the basic Volla, do you know if the Volla 22 is currently protected against the problems that the Volla has suffered about internet access and mms (not yet fully resolved) with the operator Bouygues ?

**Comment by wurzer** [comment]
It seems to be a specific problem with Bouygues. I can send and receive MMS with Deutsche Telekom, Vodafone and O2. Please check carefully the APN settings for MMS.

**Comment by gbdomubpkm** [comment]
@Pekkari RC 59 of today has broken PR-28.
 MMS no longer works !

Edit : I found a solution that works by configuring two different APNs (1 internet, 2 MMS) which takes me back two years but at least it works.

**Comment by wurzer** [comment]
MMS  service swill be shut down soon by most of the cellular network operators

**Comment by gbdomubpkm** [comment]
Hello.
I know no more updates for xenial and it isn't the good link for focal (i'm not in focal yet).
But you say mms service be shut down soon by most... Can you give links please ? I only see 'Swisscom' and 'Salt' currently.
Soon : what does it mean ?
About Bouygues, i see nothing about shut down.
Thanks.

**Comment by wurzer** [comment]
In Germany, we have three physical network operators. Telefonica has already shut down the MMS service, Deutsche Telekom will shut down MMS service at the end of this year. Vodafone didn't make an announcement, yet. 

It seems, that swisscom had also shut down this service: https://www.netzwoche.ch/news/2022-02-10/swisscom-beerdigt-mms

**Comment by gbdomubpkm** [comment]
Does this mean that, in general, the Vollas will no longer support MMS under focal and that H. W. S. no longer intends to develop on it ? Can you clarify please ? Thank you

**Comment by wurzer** [comment]
MMS ist still supported under Focal. I have checked in successfully. But if this service is or will be shut down by the mobile operators, it may not worth to investigate in corner cases. 

We realized, that Ubuntu Touch has a problem with same APN settings for Internet and MMS. 

The APN data base is even improved with Ubuntu Touch Focal. Please check this feature again, if you have switched to Ubuntu Touch Focal. I expect a version for every use with OTA-3. Currently some basic feautres are missing:

* A-GPS
* Fingerprint

Other than that, UT Focal is a real step ahead.

**Comment by gbdomubpkm** [comment]
All right. You scared me a little. It is clearer. I'm not on focal yet because OTA-1 is too basic. I expect at least OTA-2 which is not there and is long overdue. If mms had not been supported I would have stayed on xenial even longer.

**Comment by wurzer** [comment]
OTA-2 will come at the end of June. I expect a stable version for every day use with OTA-3. Is your workaround for the french mobile operator still working?

**Comment by gbdomubpkm** [comment]
As i said, i am on Xenial yet.   Yes, i can send mms but i use two different Bouygues APNs, one for internet, the other for MMS.

Will the hotspot be 100% functional on the Volla with OTA-2 focal ?

**Comment by wurzer** [comment]
I think you are the one with a SIM card, that has the same APN settings for Internet and MMS. Would you like test it on a developer or RC cahnnel?

**Comment by gbdomubpkm** [comment]
Not on focal, too immature and incomplete environment at this stage. I can't destabilize my stable custom Xenial installation to go back and forth xenial focal on my daily phone.

**Comment by wurzer** [comment]
OTA-3 will be on the staiblity level of Xenial. However as soon you can chaeck the next release, let us know, wether your SIM is working. It should be fixed now in the dev and rc and therefore also in the next stable release of UT Focal.

**Comment by gbdomubpkm** [comment]
Of course you will be informed of my tests but the bare minimum for me  is to have a functional **secure** hotspot and a functional focal Tedit app (in the OpenStore) so that I think about taking the plunge 🙂

**Comment by gbdomubpkm** [comment]
So. I have been on focal (OTA-3) since last night only 
 !! Only one APN (internet + mms) was installed and, it seems, as required. This also seems to work as it should for mms. I will confirm that everything is ok after a period of testing.
